src/libutil/args.cc

```diff
@@ -255,7 +255,13 @@ bool Args::processArgs(const Strings & args, bool finish)
         }
         if (!anyCompleted)
             exp.handler.fun(ss);
-        expectedArgs.pop_front();
+
+        processedArgs.splice(
+            processedArgs.end(),
+            expectedArgs,
+            expectedArgs.begin(),
+            ++expectedArgs.begin());
```

Comment:
This is quite roundabout (though I trust you that there's no better way). Mind adding a little comment explaining that it's morally “just” `processedArgs.emplace(expectedArgs.begin()); expectedArgs.pop_front()`?

Reply:
Good point, fixed! I used `push_back(expectedArgs.front())` now, because `emplace(expectedArgs.begin())` doesn't even compile.